python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -531,13 +537,21 @@ def _schedule_replica(
                 if scheduling_request.placement_group_strategy
                 else "PACK"
             )
-            pg = self._create_placement_group_fn(
-                scheduling_request.placement_group_bundles,
-                placement_group_strategy,
-                _soft_target_node_id=target_node_id,
-                lifetime="detached",
-                name=scheduling_request.actor_options["name"],
-            )
+            try:
+                pg = self._create_placement_group_fn(
+                    scheduling_request.placement_group_bundles,
+                    placement_group_strategy,
+                    _soft_target_node_id=target_node_id,
+                    lifetime="detached",
+                    name=scheduling_request.actor_options["name"],
+                )
+            except Exception:
+                # We add a defensive exception here, so the controller can
+                # make progress even if the placement group isn't created.
+                # See https://github.com/ray-project/ray/issues/43888.
+                logger.exception(
+                    f"Failed to create a placement group for {replica_id}."
+                )
```

Comment:
in this case we need to mark this replica as failed somehow, right?

Reply:
This happens when scheduling the replica, so there shouldn't be a replica to mark as failed. We should exit early though and leave the replica as pending. I'll add a `return`.

Reply:
We should at least return early; otherwise, this code will try to initialize the actor and remove the replica scheduling request from its internal data structure. I'll make the change.

> in this case we need to mark this replica as failed somehow, right?

Do you mean for observability purposes? Currently there is no failed state for replicas (see [`ReplicaState`](https://github.com/ray-project/ray/blob/0086eb5616f63901c5bfd5f1e57aa4b53589fb3d/python/ray/serve/_private/common.py#L108-L115)). When this scheduling code runs, the replica will be in the `STARTING` state. We could either create a new `FAILED` state, or we could leave it in the `STARTING` state indefinitely.

Reply:
If we leave the replica in the starting state, won't we basically just hang in that state forever?

Reply:
I updated the PR to set replicas to `STOPPING` and track whenever replicas fail to be scheduled.